pkg/registry/file/dynamicpathdetector/analyze_endpoints.go

```diff
@@ -0,0 +1,128 @@
+package dynamicpathdetector
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	types "github.com/kubescape/storage/pkg/apis/softwarecomposition"
+)
+
+func AnalyzeEndpoints(endpoints *[]types.HTTPEndpoint, analyzer *PathAnalyzer) ([]types.HTTPEndpoint, error) {
+	var newEndpoints []types.HTTPEndpoint
+	MergeDuplicateEndpoints(endpoints)
+	for _, endpoint := range *endpoints {
+		AnalyzeURL(endpoint.Endpoint, analyzer)
+	}
+
+	for _, endpoint := range *endpoints {
+		processedEndpoint, err := ProcessEndpoint(&endpoint, analyzer, newEndpoints)
+		if processedEndpoint == nil && err == nil || err != nil {
+			continue
+		} else {
+			newEndpoints = append(newEndpoints, *processedEndpoint)
+		}
+	}
+
+	return newEndpoints, nil
+}
+
+func ProcessEndpoint(endpoint *types.HTTPEndpoint, analyzer *PathAnalyzer, newEndpoints []types.HTTPEndpoint) (*types.HTTPEndpoint, error) {
+	url, err := AnalyzeURL(endpoint.Endpoint, analyzer)
+	if err != nil {
+		return nil, err
+	}
+
+	if url != endpoint.Endpoint {
+
+		// Check if this dynamic exists
+		for i, e := range newEndpoints {
+			if e.Endpoint == url {
+				newEndpoints[i].Methods = mergeMethods(e.Methods, endpoint.Methods)
+				newEndpoints[i].Headers = mergeHeaders(e.Headers, endpoint.Headers)
+				return nil, nil
+			}
+		}
+
+		dynamicEndpoint := types.HTTPEndpoint{
+			Endpoint:  url,
+			Methods:   endpoint.Methods,
+			Internal:  endpoint.Internal,
+			Direction: endpoint.Direction,
+			Headers:   endpoint.Headers,
+		}
+
+		return &dynamicEndpoint, nil
+	}
+
+	return endpoint, nil
+}
+
+func AnalyzeURL(urlString string, analyzer *PathAnalyzer) (string, error) {
+	if !strings.HasPrefix(urlString, "http://") && !strings.HasPrefix(urlString, "https://") {
+		urlString = "http://" + urlString
```

Comment:
maybe we want to default to https?

Reply:
It does not matter as it is used only for  parsing `parsedURL, err := url.Parse(urlString)`